src/lib/add-on/clients/inst_add-on.rb

```diff
@@ -0,0 +1,206 @@
+          end
+          InstallProduct()
+        end
+      end
+
+      # the module was started because of the kernel command line option
+      # so finish it after adding the add-ons, no UI is actually needed
+      return :auto if Installation.add_on_selected == false
+
+      @ret = RunAddOnMainDialog(
+        GetInstArgs.enable_back,
+        GetInstArgs.enable_next,
+        true,
+        Label.BackButton,
+        Label.NextButton,
+        Label.AbortButton,
+        true
+      )
+
+      if @ret == :next
+        # be careful when calling this client from other modules, this will
+        # start the workflow from the next step and THEN return back
+        # to the caller
+        @ret = ProductControl.RunFrom(
+          Ops.add(ProductControl.CurrentStep, 1),
+          true
+        )
+        @ret = :finish if @ret == :next
+      end
+
+      @ret
+
+      # EOF
+    end
+
+    # Ask to user to configure network for installing remote addons.
+    # User is aske when there is a remote add-on found and the network
+    # is not running yet.
+    #
+    # @param addon_urls list of URLs
+    # @return symbol user input result (`next, `back or `abort)
+    def NetworkSetupForAddons(addon_urls)
+      addon_urls = deep_copy(addon_urls)
+      # protocols locally available (no network needed)
+      local_protocols = ["cd", "dvd", "hd"]
+
+      # is this CD/DVD/HDD installation?
+      if Builtins.contains(
+          local_protocols,
+          Builtins.tolower(Linuxrc.InstallInf("InstMode"))
+        )
+        # is there any remote addon requiring network setup?
+        network_needed = false
+        Builtins.foreach(addon_urls) do |url|
+          # is it a remote protocol?
+          if !Builtins.contains(
+              local_protocols,
+              Builtins.tolower(Ops.get_string(URL.Parse(url), "scheme", ""))
+            )
+            network_needed = true
+            raise Break
+          end
+        end
+
+
+        if network_needed
+          # check and setup network
+          ret = Convert.to_symbol(WFM.CallFunction("inst_network_check", []))
+          Builtins.y2milestone("inst_network_check ret: %1", ret)
+
+          return ret if Builtins.contains([:back, :abort], ret)
+        end
+      end
+
+      :next
+    end
+
+    # get the URLs of the all present add-ons
+    # @return [Array<String>] list of URLs (empty list if no add-on defined)
+    def current_addons
+      AddOnProduct.add_on_products.map do |addon|
+        Pkg.SourceURL(addon["media"])
+      end
+    end
+
+    # URL schemas to be considered as CD/DVD
+    DVD_SCHEMES = ["cd", "dvd"].freeze
+
+    # Determine whether the URL points to the installation CD/DVD
+    #
+    # @param url [String] Add-on URL
+    # @return [Boolean] true if it points to the installation CD/DVD
+    def instsys_dvd?(url)
+      scheme = URL.Parse(url)["scheme"]
+      return false unless DVD_SCHEMES.include?(scheme)
```

Comment:
`URL.Parse` keeps the uppercase letters, this will not work for `DVD://` URL. Use `DVD_SCHEMES.include?(scheme.downcase)` here.